ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -501,11 +501,19 @@ func (h *FaultHandler) CheckNetworkPacketLoss() func(http.ResponseWriter, *http.
 	}
 }
 
-// decodeRequest will translate/unmarshal an incoming fault injection request into one of the network fault structs
-func decodeRequest(w http.ResponseWriter, request types.NetworkFaultRequest, requestType string, r *http.Request) error {
+// decodeRequest will log the request and then translate/unmarshal an incoming fault injection request into
+// one of the network fault structs if the request body is required.
+func decodeRequest(w http.ResponseWriter, request types.NetworkFaultRequest, requestType string,
+	r *http.Request, requiredRequestBody bool) error {
 	logRequest(requestType, r)
+
 	jsonDecoder := json.NewDecoder(r.Body)
 	if err := jsonDecoder.Decode(request); err != nil {
+		// The request has empty body. Respond an explicit message if that's required.
+		if err == io.EOF && requiredRequestBody {
```

Comment:
Does `io.EOF` always signal an empty body or could that happen for more cases such as connection issues?

As per the documentation of `http.Request` struct - 

> 	// For client requests, a nil body means the request has no
>	// body, such as a GET request. The HTTP Client's Transport
>	// is responsible for calling the Close method.

https://pkg.go.dev/net/http#Request

So should we check `r.Body == nil` instead?

Reply:
>So should we check r.Body == nil instead?
+1

Reply:
```
	// Body is the request's body.
	//
	// For client requests, a nil body means the request has no
	// body, such as a GET request. The HTTP Client's Transport
	// is responsible for calling the Close method.
	//
	// For server requests, the Request Body is always non-nil
	// but will return EOF immediately when no body is present.
	// The Server will close the request body. The ServeHTTP
	// Handler does not need to.
	//
	// Body must allow Read to be called concurrently with Close.
	// In particular, calling Close should unblock a Read waiting
	// for input.
	Body io.ReadCloser
```

Reply:
The body is always non-nil so we should not check if that is nil.

Reply:
Thanks for checking @xxx0624 . That makes sense to me.